### DISCUSSION TITLE
Using Bravo to automatically deleted unused columns?

### DISCUSSION THREAD

**Discussion by SamuelAB-VIM** [sql-bi/Bravo#713] [Q&A]
We make many reports from one complicated data model. However, a set of reports sometimes only uses 15% of the columns that are available. Our production would be much faster if we were able to purge these unused columns before distributing them.

Is there a way to do this or to provide a list of all the unused columns an a CSV/Excel file so that we can eliminate these columns using other methods ourselves? It would be easier if Bravo did this whole workflow in way we could easily automate before publishing.

**Comment by albertospelta** [comment]
Hi @SamuelAB-VIM, detecting and exporting unreferenced columns and measures in visuals is definitely on our to-do list (see #271 and #505) but we do not plan to support ability to automatically delete objects from the data model. We can provide the user with the ability to manually export a CSV/Excel list of unused columns/measures but, for a more complex scenarios, such as use in an automated background task or a [DevOps pipeline](https://www.youtube.com/watch?v=BbYqcSUXOgU), you should probably consider other tools such as [Tabular Editor](https://github.com/TabularEditor/TabularEditor) as Bravo is intended for users who don't need this level of complexity.
